MekHQ/src/mekhq/campaign/mission/AtBContract.java

```diff
@@ -1777,26 +1778,30 @@ public int calculateContractDifficulty(Campaign campaign) {
         enemyPower = (int) round(enemyPower * enemySkillMultiplier);
 
         // Estimate player power
-        int playerPower = estimatePlayerPower(campaign);
+        double playerPower = estimatePlayerPower(campaign);
 
         // Estimate the power of allied forces
-        int allyPower = 0;
-        if (!getCommandRights().isIndependent()) {
+        // TODO pull these directly from Force Generation instead of using magic numbers
+        // TODO estimate the LIAISON ratio by going through each combat lance and
+        // getting the actual average (G)BV for an allied heavy/assault mek.
+        double allyRatio = switch (getCommandRights()) {
+            case INDEPENDENT    -> 0; // no allies
```

Comment:
Now that I can see your code I'm getting a much better idea of what you're intending and I think the use of magic numbers here is perfectly reasonable. While I imagine we probably could derive a 'correct' ratio with a bunch of math, that really doesn't seem worth the effort.

Reply:
Yeah, I put that in the same category as including salvage or taking the average across all unit types instead of just meks.
Could we? Yes.
Would it be more accurate? Yes.
Would it actually change the difficulty that is displayed? Probably not.

Reply:
100% agree. When you were discussing it on discord I completely misunderstood and thought you were saying that we should use magic numbers instead of determining player power